app/packages/state/src/hooks/hooks-utils.ts

```diff
@@ -1,10 +1,56 @@
 import { useCallback, useEffect, useRef, useState } from "react";
 
-import ResizeObserver from "resize-observer-polyfill";
 import { toCamelCase } from "@fiftyone/utilities";
+import ResizeObserver from "resize-observer-polyfill";
 
 import { State, StateResolver, transformDataset, useStateUpdate } from "../";
 
+// adapted from https://usehooks.com/useLocalStorage/
+export const useBrowserStorage = <T = string>(
```

Comment:
I would propose putting this into its own named file as a default export, but just a suggestion